mayastor/src/bdev/nexus/nexus_child.rs

```diff
@@ -132,6 +132,9 @@ pub struct NexusChild {
     /// current state of the child
     #[serde(skip_serializing)]
     pub state: AtomicCell<ChildState>,
+    /// destruction started, differentiates remove events
+    #[serde(skip_serializing)]
+    destroying: AtomicCell<bool>,
```

Comment:
can we add this state into the child state as a variant?

Reply:
I used a separate "Destroying" state initially but the handling of different states in the remove handler itself seemed to mean that we'd end up with some of those states logical-ORed with the "Destroying" so I thought it was cleaner to just use a separate flag.

Reply:
Yeah, I can see its easier, and considering the short term goals appropriate, however; we can now open/share a nexus, that is being destroyed? (as we look at the state of the bdev to determine if we want to share?) We also now do `if destroying() { } else { }` which seems a whole lot less ergonomic to me than using [match](https://doc.rust-lang.org/book/ch18-03-pattern-syntax.html) statements.

Reply:
> we can now open/share a nexus, that is being destroyed?
Good point, I'll try to work in a separate "Destroying" state.

Reply:
I've changed the flag to a new ChildState. I can squash the separate commit later.